### PR TITLE
Support region-aggregation with weights-index as superset

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- ~Documentation Added~
- ~Name of contributors Added to AUTHORS.rst~
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

Per a request by @bs538, this PR extends the `aggregate_region()` method such that the computation is done if the weights-index is a superset of the data-index. Previously, the computation was only done if the index was identical.

E.g., if (for whatever reasons), CO2-prices are only given until 2050 but CO2 emissions are given until 2100, the method will now be able to compute region-aggregated CO2-prices until 2050.

The relevant change is going from `equals` to `isin` when checking the consistency of the two data- and weight-index, plus adding an element to a test. All other changes are simply clean-up of the tests and docstrings.

### CODE REVIEW

**Comment by danielhuppmann** [comment]
Note that the reduction in test coverage is due to skipping the UNFCCC integration test in the unrelated and already-merged PR #753

**Review by phackstock — COMMENTED** [review]
Just one small comment other than that, looks good to me.

File: pyam/aggregation.py [hunk 1]
```diff
@@ -214,29 +214,30 @@ def _agg_weight(data, weight, method, drop_negative_weights):
...
     if method not in ["sum", np.sum]:
-        raise ValueError("Only method 'np.sum' allowed for weighted average!")
+        raise ValueError("Only method 'np.sum' allowed for weighted average.")
 
     weight = weight.droplevel(["variable", "unit"])
 
-    if not data.droplevel(["variable", "unit"]).index.equals(weight.index):
-        raise ValueError("Inconsistent index between variable and weight!")
+    if not all(data.droplevel(["variable", "unit"]).index.isin(weight.index)):
+        raise ValueError("Inconsistent index between variable and weight.")
```
[the remark this answers is not in the document]
Comment (danielhuppmann) [reply pyam/aggregation.py:222]: Yes, this is a good idea - I'll implement it. However, that means that we'll also have [this section](https://github.com/IAMconsortium/nomenclature/blob/881d793a0fe7a1ca7996053c9d9257c68681211e/nomenclature/processor/region.py#L594) in the nomenclature package and do synchronized releases.

[the remark this answers is not in the document]
Comment (danielhuppmann) [reply pyam/aggregation.py:222]: Not  problem at all, just a reminder to you and me...

**Comment by danielhuppmann** [comment]
Closing in favor of #792

**Pull request closed without merging**